pkgs/desktops/gnome/core/gnome-software/default.nix

```
@@ -59,6 +59,12 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  # By default Flatpaks installed through gnome-software have the full Nix
+  # store path hardcoded into the launcher/desktop file. After that version
+  # of Flatpak has been garbage-collected they stop working.
+  # The same fix is already applied to the flatpak binary itself.
+  preFixup = "gappsWrapperArgs+=(--set FLATPAK_BINARY flatpak)";
```

If we were to use this workaround for now, it should be formatted as follows:

```suggestion
  preFixup = ''
    gappsWrapperArgs+=(
      # By default Flatpaks installed through gnome-software have the full Nix
      # store path hardcoded into the launcher/desktop file. After that version
      # of Flatpak has been garbage-collected they stop working.
      # The same fix is already applied to the flatpak binary itself.
      --set FLATPAK_BINARY flatpak
    )
  '';
```

And be placed according to https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md.

---

Done